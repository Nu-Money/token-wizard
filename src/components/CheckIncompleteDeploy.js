import React, { Component } from 'react'
import cancelDeploy from '../utils/cancelDeploy'
import { navigateTo } from '../utils/utils'

class CheckIncompleteDeploy extends Component {
  cancel() {
    cancelDeploy()
  }

  goToStepFour() {
    navigateTo({
      history: this.props.history,
      location: 'stepFour'
    })
  }

  render() {
    return (
      <div>
        <section className="home">
          <div className="crowdsale">
            <div className="container">
              <h1 className="title">Welcome to Token Wizard</h1>
              <p className="description">You have an incomplete deploy.</p>
              <div className="buttons">
                <button onClick={() => this.goToStepFour()} className="button button_fill">
                  Resume
                </button>
                <button onClick={() => this.cancel()} className="button button_outline">
                  Cancel
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
    )
  }
}

export default CheckIncompleteDeploy
