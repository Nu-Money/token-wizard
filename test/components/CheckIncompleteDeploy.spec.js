import React from 'react'
import renderer from 'react-test-renderer'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import { MemoryRouter } from 'react-router'
import CheckIncompleteDeploy from '../../src/components/CheckIncompleteDeploy'

configure({ adapter: new Adapter() })

describe('CheckIncompleteDeploys', () => {
  const history = { push: jest.fn() }
  it(`should render CheckIncompleteDeploys`, () => {
    // Given
    const component = renderer.create(
      <MemoryRouter initialEntries={['/']}>
        <CheckIncompleteDeploy />
      </MemoryRouter>
    )

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should navigate to StepFour`, () => {
    // Given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <CheckIncompleteDeploy history={history} />
      </MemoryRouter>
    )

    // When
    const checkIncompleteDeployComponent = wrapper.find('CheckIncompleteDeploy')
    const navigateToHandler = jest.spyOn(checkIncompleteDeployComponent.instance(), 'goToStepFour')
    wrapper.update()
    checkIncompleteDeployComponent
      .find('.buttons button')
      .at(0)
      .simulate('click')

    // Then
    expect(navigateToHandler).toHaveBeenCalledTimes(1)
  })

  it(`should cancel the deploy`, () => {
    // Given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/']}>
        <CheckIncompleteDeploy />
      </MemoryRouter>
    )

    // When
    const checkIncompleteDeployComponent = wrapper.find('CheckIncompleteDeploy')
    const cancelHandler = jest.spyOn(checkIncompleteDeployComponent.instance(), 'cancel')
    wrapper.update()
    checkIncompleteDeployComponent
      .find('.buttons button')
      .at(1)
      .simulate('click')

    // Then
    expect(cancelHandler).toHaveBeenCalledTimes(1)
  })
})
