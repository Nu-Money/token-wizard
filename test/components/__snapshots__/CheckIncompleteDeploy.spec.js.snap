// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckIncompleteDeploys should render CheckIncompleteDeploys 1`] = `
<div>
  <section
    className="home"
  >
    <div
      className="crowdsale"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          Welcome to Token Wizard
        </h1>
        <p
          className="description"
        >
          You have an incomplete deploy.
        </p>
        <div
          className="buttons"
        >
          <button
            className="button button_fill"
            onClick={[Function]}
          >
            Resume
          </button>
          <button
            className="button button_outline"
            onClick={[Function]}
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
